packages/core/src/components/tabs2/tabs2.tsx

```diff
@@ -153,6 +153,13 @@ export class Tabs2 extends AbstractComponent<ITabs2Props, ITabs2State> {
     public componentDidUpdate(_prevProps: ITabs2Props, prevState: ITabs2State) {
         if (this.state.selectedTabId !== prevState.selectedTabId) {
             this.moveSelectionIndicator();
+        } else if (prevState.selectedTabId != null) {
+            const tabs = this.getTabChildren();
+            const selectedTabIdIndex = tabs.map(child => child.props.id).indexOf(this.state.selectedTabId);
+            const prevSelectedTabIdIndex = _prevProps.children[1].map(child => child.props.id).indexOf(prevState.selectedTabId);
```

Comment:
❌  use `React.Children` for interacting with `children` arrays: https://facebook.github.io/react/docs/react-api.html#react.children

but _actually_, please do what I said previously so you can say `this.getTabChildren(prevProps.children)` instead. that's the most correct way to write this.

and add unit tests for this!!